### ISSUE TITLE
Accessability Bug - VoiceOver isn't announcing when OpenDialog is triggered from Electron

### ISSUE THREAD

**Issue by cifratila** [electron/electron#12893]
* Electron Version: Probably all
* Operating System (Platform and Version): MAC 10.9 and up
* Last known working Electron version: None

**Expected Behavior**
When triggering a OpenDialog from within Electron (let's say you have a hidden element of type file) VoiceOver should switch context to the window displayed and:
1) announce it to the user
2) let the user navigate through the contents

**Actual behavior**
When triggering a OpenDialog from within Electron, VoiceOver says it is currently on the initiating WebView entity and controls in the displayed File Picker interface are out of wack (for example you cannot go through list of files)

**To Reproduce**
```sh
$ git clone https://github.com/cifratila/electron-quick-start -b master
$ npm install
$ npm start || electron .
```

Then start VoiceOver, and you will notice the SaveDialog being announced, but not the OpenDialog (and you can also spot that there is a difference in rendering them - SaveDialog is a modal, unparented dialog, but OpenDialog is a modal parented dialog)

**Additional Information**
Current workaround was to add a build flag in electron through which the OpenDialog Pane is open in a non modal approach (the window doesn't have the Electron main window as a parent).

[mac-opendialog.zip](https://github.com/electron/electron/files/1995231/mac-opendialog.zip)
The above archive contains 2 screen videos of the current state/fixed internal state. The fixed one causes some usability issues, in terms that the OpenDialog pane can be lost if you click outside of it's frame.

If needed I can provide more details 1:1

/CC @MarshallOfSound

**Comment by BrendanMcK** [comment]
This also affects simple window.alert/confirm-style popups too.

Simple repro is to append similar to index.html:

```
    Click this: <button id='btn'>Show confirm popup</button>
    <script>
      const btn = document.getElementById('btn');
      btn.addEventListener('click', function() {
        window.confirm('button was clicked');
      });
    </script>
```

* As you tab between OK/Cancel, VO will announce the new focus,
* But VO doesn't update the outline to indicate which button has focus
* The other text in the popup isn't reachable using VO navigation

**Comment by MRayermannMSFT** [comment]
We're also running into this issue when using child `BrowserWindows`. I see the same thing that @BrendanMcK sees in the accessibility inspector. I can also see that VoiceOver is not moving to the new window, as VoiceOver's black border that shows where it's "focus" is located is still on the control that triggered the launch of the new window. That being said, normal keyboard focus did make it's way to the new window, and VoiceOver does say things when elements are focused. Unfortunately what it says on each focus change is literally everything in the new window.

**Comment by MRayermannMSFT** [comment]
Also, it looks like if you set the window's parent after showing it, you can still have VoiceOver support, and a semi modal experience. The child will not go behind the parent, but you can still do things in the parent window.

**Comment by christopherpross** [comment]
#14234

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.
